sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -614,14 +613,9 @@ object RocksDBStateStoreProvider {
 
   private val MAX_AVRO_ENCODERS_IN_CACHE = 1000
   // Add the cache at companion object level so it persists across provider instances
-  private val avroEncoderMap: NonFateSharingCache[String, AvroEncoder] = {
-    val guavaCache = CacheBuilder.newBuilder()
-      .maximumSize(MAX_AVRO_ENCODERS_IN_CACHE)  // Adjust size based on your needs
-      .expireAfterAccess(1, TimeUnit.HOURS)  // Optional: Add expiration if needed
-      .build[String, AvroEncoder]()
-
-    new NonFateSharingCache(guavaCache)
-  }
+  // TODO: SPARK-50442 - Use Guava Cache to store Avro encoders instead of concurrent
```

Comment:
nit: let's explicitly mention about the context which issue we had in the JIRA ticket, and also explicitly mention that CI PR is not sufficient; and the step to verify the new PR (running test with maven).
You don't need to change the code here but let's still ensure that JIRA has sufficient info.

Reply:
Okay updated